### PR TITLE
fix: fixed verification endpoint and validation logic for `browshot`

### LINKED ISSUE #3473: Issue with incorrect verification endpoint and validation logic for `browshot` detector
Please review the [Community Note](https://github.com/trufflesecurity/trufflehog/blob/main/.github/community_note.md) before submitting

#### Issue 1 : Correct Verification Endpoint
Verification endpoint can be fixed to point to `https://api.browshot.com/api/v1/account/info` which is aligned with the documentation to fetch the account info and also with other detector's logic.

Reference : https://browshot.com/api/documentation#account_info

#### Issue 2 : Correct validation logic for token
The key size should be 32 instead of 28.
Screenshot 2024-10-19 at 1 54 35 AM

#### Issue 1 : Incorrect Verification endpoint
Currently the API verification endpoint points to `https://api.browshot.com/api/v1/instance/list`. 

#### Issue 2 : Incorrect validation logic for token
Currently the regex is of 28 characters, which is not valid for current API keys provided by `browshot`.
Screenshot 2024-10-19 at 1 54 20 AM

## Environment
 * OS: [e.g. iOS]
 * Version [e.g. 22]

### References

* #0000

### PR DESCRIPTION
**Pull request by sahil9001**
### Description:
Fixes #3473 

Screenshot 2024-10-19 at 1 53 21 AM

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/browshot/browshot.go [hunk 1]
```diff
@@ -20,7 +21,7 @@ var (
 	client = common.SaneHttpClient()
 
 	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
-	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"browshot"}) + `\b([a-zA-Z-0-9]{28})\b`)
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"browshot"}) + `\b([a-zA-Z-0-9]{32})\b`)
```
Comment (kashifkhan0771) [inline pkg/detectors/browshot/browshot.go:24]: The token I received is 24 characters long. I couldn't find an option in the [Dashboard](https://browshot.com/dashboard#introduction) to refresh the token, so I created a second account for confirmation 😅. However, I received another API key that is also 24 characters long. I believe this needs further investigation to ensure our current regex is accurate.

**Review by kashifkhan0771 — CHANGES_REQUESTED** [review]

File: pkg/detectors/browshot/browshot.go (hunk 1 above)
Comment (ssilare-adobe) [reply pkg/detectors/browshot/browshot.go:24]: Weird, in my case it was 32, don't know the reason why exactly. They haven't even specified in the documentation.

Comment (kashifkhan0771) [reply pkg/detectors/browshot/browshot.go:24]: Any confirmation here @sahil9001?

Comment (sahil9001) [reply pkg/detectors/browshot/browshot.go:24]: I did contact them, but they didn't reach back yet.

**Comment by kashifkhan0771** [comment]
Closing this PR as there is no update from last 2 week. Feel free to reopen @sahil9001 if you get an update. Thank you!

**Pull request closed without merging**